sw/device/silicon_creator/lib/cert/dice_cwt.c

```diff
@@ -103,33 +137,177 @@ const sc_keymgr_ecc_key_t kDiceKeyCdi1 = {
 //     -2 : bstr,                   ; X coordinate, big-endian
 //     -3 : bstr                    ; Y coordinate, big-endian
 // }
+static rom_error_t cose_key_ec_p256_build(ecdsa_p256_public_key_t *uds_pubkey,
+                                          uint8_t *cose_key_buffer,
+                                          size_t *cose_key_buffer_size) {
+  struct CborOut kCborOutHandle;
+  struct CborOut *pCborOut = &kCborOutHandle;
+
+  RETURN_IF_ERROR(
+      cbor_write_out_init(pCborOut, cose_key_buffer, *cose_key_buffer_size));
+  RETURN_IF_ERROR(cbor_map_init(pCborOut, kCoseKeyMapCount));
+  RETURN_IF_ERROR(cbor_write_pair_uint_uint(pCborOut, CWT_COSE_KEY_LABEL_TYPE,
+                                            CWT_COSE_KEY_TYPE_EC2));
+  RETURN_IF_ERROR(cbor_write_pair_uint_int(pCborOut, CWT_COSE_KEY_LABEL_ALG,
+                                           CWT_COSE_KEY_ALG_ECDSA256));
+  RETURN_IF_ERROR(cbor_write_pair_int_uint(pCborOut, CWT_COSE_KEY_LABEL_CRUVE,
+                                           CWT_COSE_EC2_CRUVE_P256));
+  RETURN_IF_ERROR(cbor_write_pair_int_bytes(pCborOut, CWT_COSE_KEY_LABEL_EC2_X,
+                                            (uint8_t *)uds_pubkey->x,
+                                            sizeof(uds_pubkey->x)));
+  RETURN_IF_ERROR(cbor_write_pair_int_bytes(pCborOut, CWT_COSE_KEY_LABEL_EC2_Y,
+                                            (uint8_t *)uds_pubkey->y,
+                                            sizeof(uds_pubkey->y)));
+  *cose_key_buffer_size = CborOutSize(pCborOut);
+
+  return kErrorOk;
+}
+
 rom_error_t dice_uds_tbs_cert_build(
     hmac_digest_t *otp_creator_sw_cfg_measurement,
     hmac_digest_t *otp_owner_sw_cfg_measurement,
     hmac_digest_t *otp_rot_creator_auth_codesign_measurement,
     hmac_digest_t *otp_rot_creator_auth_state_measurement,
     cert_key_id_pair_t *key_ids, ecdsa_p256_public_key_t *uds_pubkey,
     uint8_t *tbs_cert, size_t *tbs_cert_size) {
+  HARDENED_RETURN_IF_ERROR(
+      cose_key_ec_p256_build(uds_pubkey, tbs_cert, tbs_cert_size));
+  return kErrorOk;
+}
+
+// protected:
+// a1    # map(1)
+//    01 #   unsigned(1)
+//    26 #   negative(-7)
+const uint8_t cbor_map_algo_es256[] = {0xa1, 0x01, 0x26};
```

Comment:
```suggestion
const uint8_t kDiceCwtCborMapAlgoEs256[] = {0xa1, 0x01, 0x26};
```
for style guide compliance; applies throughout: https://opentitan.org/book/doc/contributing/style_guides/c_cpp_coding_style.html#function-enum-struct-and-typedef-naming